company-id-gui/src-tauri/src/main.rs

```diff
@@ -184,7 +184,10 @@ async fn set_node_and_network(
         Net::Testnet => TESTNET_GENESIS_HASH,
     };
     if genesis_hash.to_string() != expected_genesis_hash {
-        return Err(Error::WrongNetwork(net));
+        match net {
+            Net::Mainnet => return Err(Error::WrongNetwork("Mainnet")),
+            Net::Testnet => return Err(Error::WrongNetwork("Testnet")),
```

Comment:
Wouldn't it make more sense to add a Display implementation to the `Net` type? Using thiserror for example.

Reply:
The `Net` type already has a `Display` implementation, which is what is currently printed. This implementation unfortunately displays the net code of the network instead of the name.

Reply:
Ok.